### DISCUSSION TITLE
Aura on ARM architecture

### DISCUSSION THREAD

**Discussion by mb720** [fosskers/aura#572] [Ideas]
Hi!

I like aura so much I want to run it on my Raspberry Pi :-D

I have Arch Linux up and running on my Raspberry Pi 4 using [these](https://archlinuxarm.org/platforms/armv8/broadcom/raspberry-pi-4) fine instructions.

I can see [here](https://github.com/fosskers/aura/releases), that you currently don't provide builds for the ARM architecture.

I was wondering if there are plans to produce ARM binaries and what I can do to get aura running on Raspberry Pi, respectively.

**Comment by fosskers** [comment]
Cross-compilation is somewhat difficult for Haskell. While it might take a while, you could consider compiling Aura yourself on the Pi using `stack`.

**Comment by fosskers** [comment]
Hi there, any thoughts?

**Comment by mb720** [comment]
Hi!

Since currently stack [doesn't seem to run on ARM](https://github.com/commercialhaskell/stack/issues/2103), I've put this undertaking on hold temporarily to work on a different task.

Once I get this running, I'll let you know.

**Comment by fosskers** [comment]
In that case let's leave this open until we have more to say.

**Comment by fosskers** [comment]
https://www.haskell.org/ghc/blog/20200515-ghc-on-arm.html

**Comment by fosskers** [comment]
I'll mention also that Aura doesn't *need* `stack` to compile. You could use `cabal` just fine.

**Comment by fosskers** [comment]
I have transferred this thread to Github Discussions. I'll also mention that with Aura's on-going port to Rust, ARM support will be restored, and I have made arrangements to test this with volunteers from other Arch-based distributions.
